File: Brute Force/Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry_flag = 0;
        
        for(int i = digits.size()-1; i>=0; i--){
            if(digits[i] == 9 && carry_flag != 1){
                digits[i] = 0;
                continue;
            }

            if(carry_flag == 0){
                digits[i]++;
                carry_flag = 1;
            }
        }
        if(!carry_flag){
            digits.insert(digits.begin(), 1);
        }

        return digits;
    }
};

File: Math/Plus_One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        l = len(digits) - 1
        carry = 0
        while l >= 0:
            if digits[l] < 9:
                digits[l] += 1
                return digits
            digits[l] = 0

            l -= 1

        digits.insert(0, 1)
        return digits

        
            

            

            